notification-portlet-webapp/src/main/webapp/scripts/modal-notice.js

```diff
@@ -25,7 +25,8 @@ var upmodal_notice = upmodal_notice || {};
 if (!upmodal_notice.init) {
   upmodal_notice.init = true;
 
-  (function($) {
+  (function() { // necessary?
```

Comment:
sadly yes,
otherwise `defaults`, `initNotices`, `fetchNotices`, etc. would become global variables.

Reply:
That't what I feared -- though in that case, it works differently from Java.

For readability, I would say it's a better call to put the `(function() {})();` around the `if (){}` in this case.

Reply:
EMCA Script modules will help clean code like this up, once they are supported across more browsers.

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export
Ref: https://caniuse.com/#feat=es6-module